glean_parser/schemas/metrics.1-0-0.schema.yaml

```diff
@@ -14,6 +14,10 @@ definitions:
     type: string
     pattern: "^[a-z_][a-z0-9_]*$"
 
+  dotted_snake_case:
+    type: string
+    pattern: "^[a-z_][a-z0-9_]{0,29}(\\.[a-z_][a-z0-9_]{0,29})*$"
```

Comment:
I think we should determine that.  It was 30 characters, but if we're going to use it for heirarchy, it should probably be extended.  Maybe 50?  We should be conservative because it will ultimately affect ping sizes.